client/my-sites/sharing/connections/services/google-my-business.js

```diff
@@ -26,43 +36,72 @@ export class GoogleMyBusiness extends SharingService {
-	createOrUpdateConnection = () => {};
+	createOrUpdateConnection = ( keyringConnectionId, externalUserId ) => {
+		this.props
+			.connectGoogleMyBusinessLocation( this.props.siteId, keyringConnectionId, externalUserId )
+			.catch( () => {
+				this.props.failCreateConnection( {
+					message: this.props.translate( 'Error while linking your site to %(service)s.', {
+						args: { service: this.props.service.label },
+					} ),
+				} );
+			} )
+			.finally( () => {
+				this.setState( { isConnecting: false } );
+			} );
+	};
 
 	// override `removeConnection` to remove the keyring connection instead of the publicize one
 	removeConnection = () => {
 		this.setState( { isDisconnecting: true } );
-		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+		this.props.disconnectGoogleMyBusinessLocation( this.props.siteId ).finally( () => {
+			this.setState( { isDisconnecting: false } );
+		} );
 	};
 
-	componentWillReceiveProps( { availableExternalAccounts } ) {
-		if ( ! isEqual( this.props.availableExternalAccounts, availableExternalAccounts ) ) {
-			this.setState( {
-				isConnecting: false,
-				isDisconnecting: false,
-			} );
+	componentWillMount() {
+		this.requestSettings( this.props );
+	}
+
+	requestSettings( props ) {
+		const { requestingSiteSettings, siteId } = props;
+		if ( ! requestingSiteSettings && siteId ) {
+			props.requestSiteSettings( siteId );
 		}
+	}
 
-		if ( ! this.state.isAwaitingConnections ) {
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId && this.props.siteId !== nextProps.siteId ) {
+			this.requestSettings( nextProps );
+		}
+
+		if ( this.state.isAwaitingConnections ) {
+			this.setState( {
+				isAwaitingConnections: false,
+				isSelectingAccount: true,
+			} );
 			return;
 		}
 
-		this.setState( {
-			isAwaitingConnections: false,
-			isRefreshing: false,
-		} );
+		if ( ! isEqual( this.props.brokenConnections, nextProps.brokenConnections ) ) {
+			this.setState( { isRefreshing: false } );
+		}
 
-		if ( this.didKeyringConnectionSucceed( availableExternalAccounts ) ) {
-			this.setState( { isConnecting: false } );
-			this.props.successNotice(
-				this.props.translate( 'The %(service)s account was successfully connected.', {
-					args: { service: this.props.service.label },
-					context: 'Sharing: Publicize connection confirmation',
-				} ),
-				{ id: 'publicize' }
-			);
+		// do not use `availableExternalAccounts` as a datasource to know if a connection was successful
+		// because we allow the same location to be used on multiple sites in the case of GMB.
+		// Just check that a new connection is added
+		if ( ! isEqual( this.props.siteUserConnections, nextProps.siteUserConnections ) ) {
+			this.setState( {
+				isConnecting: false,
+				isDisconnecting: false,
+			} );
 		}
 	}
 
+	getDisclamerText() {
+		return this.props.translate( 'Select the location you wish to connect your site to.' );
```

Comment:
I'm slightly concerned that this won't work well when there is only a single location available as you cannot select anything. Maybe `Confirm the location you wish to connect your site to.` would work better?

Reply:
Good point 👍